compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/DesktopPopup.desktop.kt

```diff
@@ -44,52 +51,97 @@ fun rememberCursorPositionProvider(
     alignment: Alignment = Alignment.BottomEnd,
     windowMargin: Dp = 4.dp
 ): PopupPositionProvider = with(LocalDensity.current) {
-    val component = LocalLayerContainer.current
-    val cursorPoint = remember {
-        val awtMousePosition = MouseInfo.getPointerInfo().location
-        convertPointFromScreen(awtMousePosition, component)
-        IntOffset(
-            (awtMousePosition.x * component.density.density).toInt(),
-            (awtMousePosition.y * component.density.density).toInt()
-        )
-    }
+    val windowMarginPx = windowMargin.roundToPx()
     val offsetPx = IntOffset(offset.x.roundToPx(), offset.y.roundToPx())
+    val cursorPosition = rememberCursorPosition()
+
+    PopupPositionProviderAtOffset(
+        anchorPx = cursorPosition,
+        isAnchorRelative = false,
+        offsetPx = offsetPx,
+        alignment = alignment,
+        windowMarginPx = windowMarginPx
+    )
+}
+
+/**
+ * A [PopupPositionProvider] that positions the popup at the given position relative to the anchor.
+ *
+ * @param positionPx the offset, in pixels, relative to the anchor, to position the popup at.
+ * @param offset [DpOffset] to be added to the position of the popup.
+ * @param alignment The alignment of the popup relative to desired position.
+ * @param windowMargin Defines the area within the window that limits the placement of the popup.
+ */
+@Composable
+fun popupPositionProviderAtPosition(
+    positionPx: IntOffset,
+    offset: DpOffset = DpOffset.Zero,
+    alignment: Alignment = Alignment.BottomEnd,
+    windowMargin: Dp = 4.dp
+): PopupPositionProvider = with(LocalDensity.current) {
     val windowMarginPx = windowMargin.roundToPx()
-    object : PopupPositionProvider {
-        override fun calculatePosition(
-            anchorBounds: IntRect,
-            windowSize: IntSize,
-            layoutDirection: LayoutDirection,
-            popupContentSize: IntSize
-        ) = with(density) {
-            val anchor = IntRect(cursorPoint, IntSize.Zero)
-            val tooltipArea = IntRect(
-                IntOffset(
-                    anchor.left - popupContentSize.width,
-                    anchor.top - popupContentSize.height,
-                ),
-                IntSize(
-                    popupContentSize.width * 2,
-                    popupContentSize.height * 2
-                )
+    val offsetPx = positionPx + IntOffset(offset.x.roundToPx(), offset.y.roundToPx())
+
+    PopupPositionProviderAtOffset(
```

Comment:
Because it is Composable, we should add `remember`, in case we will have state in the `PopupPositionProviderAtOffset` in the future.

Reply:
The only reason it needs to be `@Composable` is because it needs to use `LocalDensity.current` (which we may remove, due to the previous point of discussion). Either way, I don't think it will ever need to remember anything.